### ISSUE TITLE
Add baseUrl to a importmap()

### ISSUE THREAD

**Issue by tacman** [StenopePHP/Stenope#173]
Symfony's importmap() function generates

```
<script type="importmap">
{
    "imports": {
        "app": "/assets/app-e4599d4ad24d45913e4d05569a4414b4.js",
        "/assets/bootstrap.js": "/assets/bootstrap-c423b8bbc1f9cae218c105ca8ca9f767.js",
        "@symfony/stimulus-bundle": "/assets/@symfony/stimulus-bundle/loader-e1ee9ace0562f2e6a52301e4ccc8627d.js",
        "/assets/@symfony/stimulus-bundle/controllers.js": "/assets/@symfony/stimulus-bundle/controllers-abf606f3085d4a46150023eb97dbf33b.js",
        "/assets/controllers/hello_controller.js": "/assets/controllers/hello_controller-c19dd529e97e8ef33363253ca96d13fe.js",
        "/assets/controllers/api_controller.js": "/assets/controllers/api_controller-192ad26be8b939ca8ca28b3dad1fa53f.js",
        "@hotwired/stimulus": "https://cdn.jsdelivr.net/npm/@hotwired/stimulus@3.2.2/+esm",
        "@fullcalendar/daygrid": "https://cdn.jsdelivr.net/npm/@fullcalendar/daygrid@6.1.9/+esm",
        "@fullcalendar/interaction": "https://cdn.jsdelivr.net/npm/@fullcalendar/interaction@6.1.9/+esm",
        "@fullcalendar/core": "https://cdn.jsdelivr.net/npm/@fullcalendar/core@6.1.9/+esm"
    }
}
```

when there's a base url, though, those routes need to be prefixed, like the path() commands.

I use this to publish to github pages:

bin/console -e prod stenope:build ./docs --base-url /micro-calendar/

https://tacman.github.io/micro-calendar/fullcalendar/

I've hacked around this problem by embedding my app.js in the HTML itself, but obviously it'd be better if the generator tweaked the paths.

Thanks.

**Comment by ogizanagi** [comment]
Is there a way to provide a base URL when using importmap? I suspect this to be more a Symfony feature request rather than a Stenope one

**Comment by tacman** [comment]
Not that I know of.  Can you point to where in your code the links are created?  

Your code IS working as expected for the web page links, just not for the assets.  I think the issue is best fixed in Stenope, since --base-url is being passed into your build command.  The asset links are probably generated differently than the embedded HTML page links. 

For production, the assetmapper files exist on disk, so it's just a simple path (in dev, there's some magic that maps the asset routes).  That's why I _think_ the baseUrl needs to be added in stenope.

I can provide a very simple demo repo if that's helpful.

**Comment by ogizanagi** [comment]
I'm not used to the AssetMapper yet. That's why I'm wondering how could Stenope communicate the baseURL to generate the proper links when using importmap? Were should we hook?

For routing (`path`), this is handled by setting the request context:

https://github.com/StenopePHP/Stenope/blob/4ff235c0db3bfb3c3a934058b3a5af4142d82161/src/Builder.php#L186-L189

What would be the equivalent for generating the proper importmap?
Or is there a config entry that would allow to configure the base URL in the userland code, allowing to use an env var?

**Comment by tacman** [comment]
Thanks.  AssetMapper 6.4 is marked as experimental, but will be stable in 6.4.  I'll add this as an issue, as it does seem like baseUrl should be honored there, too.  Can we keep this open until it's functional?

I've been playing around with the steno-skeleton, trying to remove webpack and replace it with AssetMapper.  It'll be very cool when it's done!

**Comment by ogizanagi** [comment]
It seems to me the [default asset package](https://github.com/symfony/symfony/blob/de3d115496c0ab76d9849896e9d4c8a15b91eb1f/src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php#L30) is [used to render the `importmap`](https://github.com/symfony/symfony/blob/de3d115496c0ab76d9849896e9d4c8a15b91eb1f/src/Symfony/Component/AssetMapper/ImportMap/ImportMapRenderer.php#L34). Did you try to set the [`base_url` here](https://symfony.com/doc/current/reference/configuration/framework.html#base-path) (using an env var to provide when building for your production environment) ?

**Comment by ogizanagi** [comment]
Should work using 6.4 at least : https://github.com/symfony/symfony/issues/52561

**Comment by ogizanagi** [comment]
Closing as resolved, using Symfony 6.4/7.0
